### PR TITLE
Docs and examples for packages

### LINKED ISSUE #187: [TASK] Update README and documentation
## Overview

We need to improve the documentation which includes user instructions, usage examples, and removing dead links. 

- [x] Update `README.md`: remove dead links, include additional usage examples (e.g. connect to substrate chain), testing instructions
- [ ] Add instructions and description to [wasm-utils](https://github.com/webb-tools/webb.js/tree/master/packages/wasm-utils) 
- [ ] Add inline documentation to all applicable functions  
- [x] Add additional examples
- [x] Add `Type doc` and linting && CI Action for docs

### PR DESCRIPTION
**Pull request by AhmedKorim**
Closes #187

### CODE REVIEW

File: packages/api-providers/src/abstracts/anchor/anchor-deposit.ts [hunk 1]
```diff
@@ -6,10 +6,14 @@ import { WebbApiProvider } from '../webb-provider.interface.js';
 
 export type AnchorSize = MixerSize;
 
-export abstract class AnchorDeposit<
-  T extends WebbApiProvider<any>,
-  K extends DepositPayload = DepositPayload<any>
-> extends MixerDeposit<T, K> {
+// Todo: should we extract the interface of MixerDeposit on another class and rename `generateBridgeNote` to generate note?
```
Comment (nepoche) [inline packages/api-providers/src/abstracts/anchor/anchor-deposit.ts:9]:
```suggestion
// Todo: should we extract the interface of MixerDeposit on another class and rename `generateBridgeNote` to generate note
```
Agreed this refactor should take place - especially with the removal of tornado.

File: packages/api-providers/src/abstracts/mixer/mixer-withdraw.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ export enum WithdrawState {
 
   SendingTransaction, // There is a withdraw in progress, and it's on the step Sending the Transaction whether directly or through relayers
 
-  Done, // the withdraw is Done and seceded , the next tic the instance should be ideal
+  Done, // the withdraw is Done and seceded, the next tic the instance should be ideal
```
Comment (nepoche) [inline packages/api-providers/src/abstracts/mixer/mixer-withdraw.ts:20]:
```suggestion
  Done, // the withdraw is Done and succeeded, the next tic the instance should be ideal
```

File: packages/api-providers/src/abstracts/relayer/webb-relayer.class.ts [hunk 3]
```diff
@@ -89,16 +98,20 @@ export interface RelayerInfo {
...
- *  Webb relayers manager
- *  this will fetch/mange/provide this relayers and there capabilities
+ * Webb relayers manager
+ * this will fetch/mange/provide this relayers and there capabilities
  *
- * */
+ * @param capabilities - storage for relayers capabilities
+ * @param relayerConfigs - The whole relayers configuration of the project
+ * @param chainNameAdapter - An adapter for getting the  InternalChainId of the chain name and the base
+ * @param appConfig - App config is used for misc pursues
```
Comment (nepoche) [inline packages/api-providers/src/abstracts/relayer/webb-relayer.class.ts:107]:
```suggestion
 * @param appConfig - App config is used for looking up configuration values for issuing queries on the relayers
```

File: packages/api-providers/src/abstracts/relayer/webb-relayer.class.ts [hunk 4]
```diff
@@ -298,28 +313,55 @@ export class WebbRelayerBuilder {
...
   CleanExit,
-  /// failed to create the withdraw
   Errored
 }
 
+/**
+ * Fetching relayer from the relayer is faster that from node/contract
+ * it will return the whole leaves, And what block number is used in the last pulling of leaves
+ *
+ * @param leaves - Array of hex representation of the leave
```
Comment (nepoche) [inline packages/api-providers/src/abstracts/relayer/webb-relayer.class.ts:337]:
```suggestion
 * @param leaves - Array of hex representation of the leaves
```

File: packages/api-providers/src/abstracts/relayer/webb-relayer.class.ts [hunk 5]
```diff
@@ -298,28 +313,55 @@ export class WebbRelayerBuilder {
...
   Continue,
-  /// the withdraw is done with success
   CleanExit,
-  /// failed to create the withdraw
   Errored
 }
 
+/**
+ * Fetching relayer from the relayer is faster that from node/contract
+ * it will return the whole leaves, And what block number is used in the last pulling of leaves
```
Comment (nepoche) [inline packages/api-providers/src/abstracts/relayer/webb-relayer.class.ts:335]:
```suggestion
 * Fetching leaves from the relayer is faster than querying a chain's node.
 * The relayer will return it's state for the given merkle tree - all of the leaves up to the latest synced block value.
```

File: packages/api-providers/src/abstracts/webb-provider.interface.ts [hunk 6]
```diff
@@ -119,58 +122,74 @@ export type TXNotificationPayload<T = undefined> = {
...
 
+/**
+ * The representation of an api provider
+ *
+ * @param accounts - Accounts Adapter will have all methods related to the provider accounts.
+ * @param methods - All of the available methods  of the API provider.
+ * @param destroy -  A hook will be called to drop the provider and do cleanup listeners etc.
+ * @param capabilities - Manifesto of the supported actions of the provider.
+ * @param endSession - Clean up for the provider that will remove the side effects.
+ * @param relayingManager - Clean up for the provider that will remove the side effects
```
Comment (nepoche) [inline packages/api-providers/src/abstracts/webb-provider.interface.ts:177]:
```suggestion
 * @param relayingManager - Object used by the provider for sending transactions or queries to a compatible relayer.
```

File: packages/api-providers/src/abstracts/webb-provider.interface.ts [hunk 7]
```diff
@@ -119,58 +122,74 @@ export type TXNotificationPayload<T = undefined> = {
...
+ * @param getProvider - A getter method for getting the underline provider
```
Comment (nepoche) [inline packages/api-providers/src/abstracts/webb-provider.interface.ts:178]:
```suggestion
 * @param getProvider - A getter method for getting the underlying provider
```

File: packages/sdk-core/README.md [hunk 8]
```diff
@@ -1,3 +1,55 @@
 # @webb-tools/sdk-core
 
-A collection of Typescript classes and utilities for Webb. 
\ No newline at end of file
+#  helpers for Zero-Knowledge and deposit notes it uses `wasm-utils`, and provider easier API
```
Comment (nepoche) [inline packages/sdk-core/README.md:3]:
```suggestion
#  Helpers for Zero-Knowledge and Note generation
The package provides a cleaner API to the underlying `wasm-utils`
```

File: packages/sdk-core/README.md [hunk 9]
```diff
@@ -1,3 +1,55 @@
...
+
+# Deposit note
+The note contain the data.
+For the browser it runs inside wasm
```
Comment (nepoche) [inline packages/sdk-core/README.md:7]:
```suggestion
The note contains data about a deposit
```

**Review by nepoche — CHANGES_REQUESTED** [review]
Nice documentation and examples! Just a few comments - let me know if there are any that you don't agree with.

File: examples/node/sdk-core/mixer.ts [hunk 10]
```diff
@@ -0,0 +1,132 @@
+import {Note, NoteGenInput, ProvingManager, u8aToHex} from '@webb-tools/sdk-core/build/index.js';
+import {fetchRPCTreeLeaves, polkadotTx, preparePolkadotApi, transferBalance} from "../utils.js";
+import {decodeAddress, Keyring} from "@polkadot/keyring";
+import {cryptoWaitReady} from '@polkadot/util-crypto'
+import {ProvingManagerSetupInput} from "@webb-tools/sdk-core/src";
```
[the remark this answers is not in the document]
Comment (AhmedKorim) [reply examples/node/sdk-core/mixer.ts:5]: I agree if it works the same way

**Comment by dutterbutter** [comment]
I will follow up this PR with a subsequent pr that deploys typedocs to gh:pages.

**Pull request merged**